Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+    def default(self, obj):
+        if isinstance(obj, datetime):
+            return obj.strftime('%Y-%m-%dT%H:%M:%S')
+        elif isinstance(obj, date):
+            return obj.strftime('%Y-%m-%d')
+        # Let the base class default method raise the TypeError
+        return json.JSONEncoder.default(self, obj)
+
+
+def parse_resource_ids(resource_id):
+    id_list = resource_id.replace(" ", "")
+    resourceIds = id_list.split(",")
+    return resourceIds
+
+
+def multi_split(data):
+    data = data.replace(" ", "")
+    data = data.split(";")
+    return data
+
+
+def parse_date(dt):
+    try:
+        arr = dt.split("-")
+        parsed_date = (datetime(int(arr[0]), int(arr[1]), int(arr[2]))).isoformat()
+    except ValueError as e:
+        return_error("Date could not be parsed. Please check the date again.\n{error}".format(error=type(e)))
+    return parsed_date
+
+
+"""MAIN FUNCTIONS"""
+
+
+def describe_regions_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    data = []
+    kwargs = {}
+    if args.get('regionNames') is not None:
+        kwargs.update({'RegionNames': parse_resource_ids(args.get('regionNames'))})
+
+    response = client.describe_regions(**kwargs)
+    for region in response['Regions']:
+        data.append({
+            'Endpoint': region['Endpoint'],
+            'RegionName': region['RegionName']
+        })
+
+    ec = {'AWS.Regions(val.RegionName === obj.RegionName)': data}
+    human_readable = tableToMarkdown('AWS Regions', data)
+    return_outputs(human_readable, ec)
+
+
+def describe_instances_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    obj = vars(client._client_config)
+    data = []
+    kwargs = {}
+    output = []
+    if args.get('filters') is not None:
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('instanceIds') is not None:
+        kwargs.update({'InstanceIds': parse_resource_ids(args.get('instanceIds'))})
+
+    response = client.describe_instances(**kwargs)
+    for i, reservation in enumerate(response['Reservations']):
+        for instance in reservation['Instances']:
+            data.append({
+                'InstanceId': instance['InstanceId'],
+                'ImageId': instance['ImageId'],
+                'State': instance['State']['Name'],
+                'PublicIPAddress': instance.get('PublicIpAddress'),
+                'Region': obj['_user_provided_options']['region_name'],
+                'Type': instance['InstanceType'],
+                'LaunchDate': datetime.strftime(instance['LaunchTime'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.